cmd/keeper/keeper.go

```diff
@@ -1754,3 +1757,86 @@ func keeper(c *cobra.Command, args []string) {
 
 	lockFile.Close()
 }
+
+// IsMaster return if the db is the cluster master db.
+// A master is a db that:
+// * Has a master db role
+// or
+// * Has a standby db role with followtype external
+func IsMaster(db *cluster.DB) bool {
+	switch db.Spec.Role {
+	case common.RoleMaster:
+		return true
+	case common.RoleStandby:
+		if db.Spec.FollowConfig.Type == cluster.FollowTypeExternal {
+			return true
+		}
+		return false
+	default:
+		panic("invalid db role in db Spec")
+	}
+}
+
+// generate default HBA depending on DefaultSUReplAccessMode.
+func (p *PostgresKeeper) generateHBA(cd *cluster.ClusterData, db *cluster.DB) []string {
+
+	// Minimal entries for local normal and replication connections needed by the stolon keeper
+	// Matched local connections are for postgres database and suUsername user with md5 auth
+	// Matched local replication connections are for replUsername user with md5 auth
+	computedHBA := []string{
+		fmt.Sprintf("local postgres %s %s\n", p.pgSUUsername, p.pgSUAuthMethod),
+		fmt.Sprintf("local replication %s %s\n", p.pgReplUsername, p.pgReplAuthMethod),
+	}
+
+	switch *cd.Cluster.DefSpec().DefaultSUReplAccessMode {
+	// Case all access mode
+	case cluster.SUReplAccessAll:
+
+		computedHBA = append(
+			computedHBA,
+			fmt.Sprintf("host all %s %s %s", p.pgSUUsername, "0.0.0.0/0", p.pgSUAuthMethod),
+			fmt.Sprintf("host all %s %s %s", p.pgSUUsername, "::0/0", p.pgSUAuthMethod),
+			fmt.Sprintf("host replication %s %s %s", p.pgReplUsername, "0.0.0.0/0", p.pgReplAuthMethod),
+			fmt.Sprintf("host replication %s %s %s", p.pgReplUsername, "::0/0", p.pgReplAuthMethod),
+		)
+
+	// Case strict access mode
+	case cluster.SUReplAccessStrict:
+		// allow for all servers su access
+		// we should not allow replication acces for 127.0.0.1 since it can be then forwarded through proxy.
```

Comment:
Sorry but I forgot why this is needed. The comment isn't really clear and doesn't looks related to the added rules.

Reply:
True,  it's changed